File: library/src/main/java/com/getstream/sdk/chat/enums/QuerySort.java
package com.getstream.sdk.chat.enums;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class QuerySort {
    private static int DESC = -1;
    private static int ASC = 1;
    private List<Map<String, Object>> mSort;

    public List<Map<String, Object>> getData() {
        return mSort;
    }

    public QuerySort clone() {
        QuerySort _this = new QuerySort();
        if (mSort == null) {
            mSort = new ArrayList<>();
        }
        _this.mSort = new ArrayList<>(mSort);
        return _this;
    }

    private QuerySort add(String fieldName, Number direction) {
        Map<String, Object> v = new HashMap<>();
        v.put("field", fieldName);
        v.put("direction", direction);
        QuerySort _this = clone();
        _this.mSort.add(v);
        return _this;
    }

    public QuerySort asc(String fieldName) {
        return add(fieldName, ASC);
    }

    public QuerySort desc(String fieldName) {
        return add(fieldName, DESC);
    }

}

File: library/src/main/java/io/getstream/chat/android/client/api/models/QuerySort.kt
package io.getstream.chat.android.client.api.models


data class QuerySort(val data: MutableList<Map<String, Any>> = mutableListOf()) {

    private fun add(fieldName: String, direction: Int): QuerySort {
        val map = mutableMapOf<String, Any>()
        map["field"] = fieldName
        map["direction"] = direction
        data.add(map)
        return this
    }

    fun asc(field: String): QuerySort {
        return add(field, ASC)
    }

    fun desc(field: String): QuerySort {
        return add(field, DESC)
    }

    companion object {
        private const val DESC = -1
        private const val ASC = 1
    }
}



